forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs

```diff
@@ -0,0 +1,245 @@
+
+    #endregion
+
+    #region Public Properties
+
+    public override DicomVR ValueRepresentation => DicomVR.DS;
+
+    #endregion
+
+    #region Public Members
+
+    public static string ToDecimalString(decimal value)
+    {
+        var valueString = value.ToString(CultureInfo.InvariantCulture);
+        if (valueString.Length > 16)
+        {
+            valueString = value.ToString("G11", CultureInfo.InvariantCulture);
+        }
+        return valueString;
+    }
+
+    #endregion
+
+}
+
+/// <summary>Integer String (IS)</summary>
+public class DicomIntegerString : DicomMultiStringOrNumberElement<int>
+{
+    #region Public Constructors
+
+    public DicomIntegerString(DicomTag tag, params int[] values)
+        : base(tag,
+              new Func<int, string>(x => x.ToString(CultureInfo.InvariantCulture)),
+              new Func<string, int>(x => int.Parse(x, NumberStyles.Integer | NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture)),
+              values)
+    {
+    }
+
+    public DicomIntegerString(DicomTag tag, params string[] values)
+        : base(tag, values)
+    {
+    }
+
+    public DicomIntegerString(DicomTag tag, IByteBuffer data)
+        : base(tag, data)
+    {
+    }
+
+    #endregion
+
+    #region Public Properties
+
+    public override DicomVR ValueRepresentation => DicomVR.IS;
+
+    #endregion
+}
+
+/// <summary>Signed Very Long (SV)</summary>
+public class DicomSignedVeryLong : DicomMultiStringOrNumberElement<long>
+{
+    #region Public Constructors
+
+    public DicomSignedVeryLong(DicomTag tag, params long[] values)
+        : base(tag,
+              new Func<long, string>(x => x.ToString()),
+              new Func<string, long>(x => long.Parse(x, NumberStyles.Any, CultureInfo.InvariantCulture)),
+              values)
+    {
+    }
+
+    public DicomSignedVeryLong(DicomTag tag, IByteBuffer data)
+        : base(tag, data)
+    {
+    }
+
+    #endregion
+
+    #region Public Properties
+
+    public override DicomVR ValueRepresentation => DicomVR.SV;
+
+    #endregion
+}
+
+/// <summary>Unsigned Very Long (UV)</summary>
+public class DicomUnsignedVeryLong : DicomMultiStringOrNumberElement<ulong>
+{
+    #region Public Constructors
+
+    public DicomUnsignedVeryLong(DicomTag tag, params ulong[] values)
+        : base(tag,
+              new Func<ulong, string>(x => x.ToString()),
+              new Func<string, ulong>(x => ulong.Parse(x, NumberStyles.Any, CultureInfo.InvariantCulture)),
+              values)
+    {
+    }
+
+    public DicomUnsignedVeryLong(DicomTag tag, IByteBuffer data)
+        : base(tag, data)
+    {
+    }
+
+    #endregion
+
+    #region Public Properties
+
+    public override DicomVR ValueRepresentation => DicomVR.UV;
+
+    #endregion
+}
+
+/// <summary>
+/// Base class to handle Multi String/Number VR Types
+/// e.g. DS, IS, SV, and UV
+/// </summary>
+public abstract class DicomMultiStringOrNumberElement<TType> : DicomMultiStringElement where TType : struct
```

Comment:
Why `TType` instead of `T`?

Reply:
"Get" override already uses "T", which is different from TType at the class level.

For e.g., the purpose of "T" in "Get" is to be able to ask for "Get<int>" on a VeryLong dicom element.